feat(core): Can access component optional props (#3892)
refinedev/refine: documentation/docs/migration-guide/router-provider.md

### Behavioral Changes in Routing

Since **refine** doesn't create routes internally anymore, you are free to create your routes according to your framework without any limitation.

This means, tasks like `authentication` and `authorization` are also de-coupled from **refine**.

However, for convenience we still provide a set of helpers for you to handle these tasks easily.

#### Authentication and Access Control

> This also means that the authentication and access control redirections are handled according to your framework.

Reviewer: ```suggestion
This also means that the authentication and access control flows should be handled according to your framework.
```

However **refine** still provides some helpers for user's convenience.

For **authentication**, you can use [`Authenticated`](/docs/api-reference/core/components/auth/authenticated) component as wrapper or use `useIsAuthenticated` hook inside your components.

> For **authorization**, you can use [`CanAccess`](/docs/api-reference/core/components/accessControl/canAccess) components as wrapper or use `useCan` hook in your components.

Reviewer: is this the one link causing error on build?
